packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -3470,13 +3467,10 @@ mixin DiagnosticableTreeMixin implements DiagnosticableTree {
   }
 
   @override
-  String? toStringShallow({
+  String toStringShallow({
     String joiner = ', ',
     DiagnosticLevel minLevel = DiagnosticLevel.debug,
   }) {
-    if (kReleaseMode) {
```

Comment:
IIRC the reason these were added is that dart2js could not optimize the function otherwise. If that is still the case I would expect  it to show up on the web size benchmarks

Reply:
Based on the history, @dnfield added the original `if (kReleaseMode) return` logic, then @ferhatb added the `assert()` around the body. This removes the if condition and leaves the assert so that the method can be non-nullable. I think it was ferhat's change that improved size, and this keeps that logic. Let's keep an eye on the numbers though, certainly...

Reply:
Ahh, yeah - I think I was remembering it backwards

Reply:
The problem with making this an assert is that profile mode will no longer show as much helpful information. But I guess we don't have any tests covering that and no one is complaining about it so oh well.

Reply:
My general opinion on this is release and profile should be as close to identical as possible except for the minimum required to actually enable collecting metrics.